internal/handlers/common/projection.go

```diff
@@ -392,35 +420,51 @@ func includeProjection(path types.Path, source any, projected *types.Document) (
 // with the key to remove that document. This is not the case in document.Remove(key).
 // Dot notation with array index path do not exclude unlike document.RemoveByPath(key).
 //
-//	Examples: "v.foo" path exclusion projection:
+// Returned command error code:
+//
+//   - ErrExclusionPositionalProjection when positional projection `$` is used.
+//
+// Examples: "v.foo" path exclusion projection:
+//
 //	{v: {foo: 1}}                       -> {v: {}}
 //	{v: {foo: 1, bar: 1}}               -> {v: {bar: 1}}
 //	{v: [{foo: 1}, {foo: 2}]}           -> {v: [{}, {}]}
 //	{v: [{foo: 1}, {foo: 2}, {bar: 1}]} -> {v: [{}, {}, {bar: 1}]}
 //
-//	Example: "v.0.foo" path exclusion projection:
+// Example: "v.0.foo" path exclusion projection:
+//
 //	{v: [{foo: 1}, {foo: 2}]}           -> {v: [{foo: 1}, {foo: 2}]}
-func excludeProjection(path types.Path, projected any) {
+func excludeProjection(path types.Path, projected any) error {
 	key := path.Prefix()
 
+	if key == "$" {
```

Comment:
Maybe we can move this check to `ValidateProjection`?

Reply:
Yes definitely, I've moved them now. And discovered more errors. It was actually the hardest part. Because I didn't figure out the logical order of errors, hence few repeated ones.

Reply:
Since you moved that check to `ValidateProjection`, can we remove it here?

Reply:
Thanks yes!